internal/pkg/auth/storage_test.go

```diff
@@ -452,16 +606,191 @@ func TestSetGetAuthFieldEncodedTextFile(t *testing.T) {
+			authFlow:        AUTH_FLOW_SERVICE_ACCOUNT_KEY,
+			expectedEmail:   "test@test.com",
+		},
+		{
+			description:   "custom profile, user token",
+			activeProfile: "test-profile",
+			userEmail:     "test@test.com",
+			authFlow:      AUTH_FLOW_USER_TOKEN,
+			expectedEmail: "test@test.com",
+		},
+		{
+			description:     "custom profile, service acc token",
+			activeProfile:   "test-profile",
+			serviceAccEmail: "test@test.com",
+			authFlow:        AUTH_FLOW_SERVICE_ACCOUNT_TOKEN,
+			expectedEmail:   "test@test.com",
+		},
+		{
+			description:     "custom profile, service acc key",
+			activeProfile:   "test-profile",
+			serviceAccEmail: "test@test.com",
+			authFlow:        AUTH_FLOW_SERVICE_ACCOUNT_KEY,
+			expectedEmail:   "test@test.com",
+		},
+		{
+			description:   "no email, user token",
+			activeProfile: "test-profile",
+			authFlow:      AUTH_FLOW_USER_TOKEN,
+			expectedEmail: "",
+		},
+		{
+			description:   "no email, service acc token",
+			activeProfile: "test-profile",
+			authFlow:      AUTH_FLOW_SERVICE_ACCOUNT_TOKEN,
+			expectedEmail: "",
+		},
+		{
+			description:   "no email, service acc key",
+			activeProfile: "test-profile",
+			authFlow:      AUTH_FLOW_SERVICE_ACCOUNT_KEY,
+			expectedEmail: "",
+		},
+		{
+			description:   "user not authenticated",
+			activeProfile: "test-profile",
+			expectedEmail: "",
+		},
+		{
+			description:     "both emails, user not authenticated",
+			activeProfile:   "test-profile",
+			userEmail:       "test@test.com",
+			serviceAccEmail: "test2@test.com",
+			expectedEmail:   "",
+		},
+		{
+			description:     "both emails, user token",
+			activeProfile:   "test-profile",
+			userEmail:       "test@test.com",
+			serviceAccEmail: "test2@test.com",
+			authFlow:        AUTH_FLOW_USER_TOKEN,
+			expectedEmail:   "test@test.com",
+		},
+		{
+			description:     "both emails, service account token",
+			activeProfile:   "test-profile",
+			userEmail:       "test@test.com",
+			serviceAccEmail: "test2@test.com",
+			authFlow:        AUTH_FLOW_SERVICE_ACCOUNT_TOKEN,
+			expectedEmail:   "test2@test.com",
+		},
+		{
+			description:     "both emails, service account key",
+			activeProfile:   "test-profile",
+			userEmail:       "test@test.com",
+			serviceAccEmail: "test2@test.com",
+			authFlow:        AUTH_FLOW_SERVICE_ACCOUNT_KEY,
+			expectedEmail:   "test2@test.com",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			keyring.MockInit()
+
+			// Append random string to profile name to avoid conflicts
+			tt.activeProfile = makeProfileNameUnique(tt.activeProfile)
+
+			// Make sure profile name is valid
+			err := config.ValidateProfile(tt.activeProfile)
+			if err != nil {
+				t.Fatalf("Profile name \"%s\" is invalid: %v", tt.activeProfile, err)
+			}
+
+			oldUserEmail, _ := getAuthFieldWithProfile(tt.activeProfile, USER_EMAIL)
+			oldServiceAccEmail, _ := getAuthFieldWithProfile(tt.activeProfile, SERVICE_ACCOUNT_EMAIL)
```

Comment:
Is this logic to reset previous values still needed now that we are using a mocked keyring?

Reply:
No, nice catch